src/modules/services-systemd/main.py

```diff
@@ -23,96 +23,64 @@
-            mandatory = svc.get("mandatory", False)
-
-        ec = libcalamares.utils.target_env_call(
-            ['systemctl', command, "{}{}".format(name, suffix)]
-            )
-
-        if ec != 0:
+            if not unit.has_key("name"):
+                libcalamares.utils.error("The key 'name' is missing from the mapping {_unit!s}. Continuing to the next unit.".format(_unit=str(unit)))
+                continue 
+            name = unit["name"]
+            action = unit.get("action", "enable")
+            mandatory = unit.get("mandatory", False)
+
+        exit_code = libcalamares.utils.target_env_call(
+            ['systemctl', action, name]
+        )
+
+        if exit_code != 0:
             libcalamares.utils.warning(
-                "Cannot {} systemd {} {}".format(command, suffix, name)
-                )
+                "Cannot {} systemd unit {}".format(action, name)
+            )
             libcalamares.utils.warning(
-                "systemctl {} call in chroot returned error code {}".format(command, ec)
-                )
+                "systemctl {} call in chroot returned error code {}".format(action, exit_code)
+            )
             if mandatory:
-                title = _("Cannot modify service")
-                diagnostic = _("<code>systemctl {arg!s}</code> call in chroot returned error code {num!s}.").format(arg=command, num=ec)
-
-                if command == "enable" and suffix == ".service":
-                    description = _("Cannot enable systemd service <code>{name!s}</code>.")
-                elif command == "enable" and suffix == ".target":
-                    description = _("Cannot enable systemd target <code>{name!s}</code>.")
-                elif command == "enable" and suffix == ".timer":
-                    description = _("Cannot enable systemd timer <code>{name!s}</code>.")
-                elif command == "disable" and suffix == ".service":
-                    description = _("Cannot enable systemd service <code>{name!s}</code>.")
-                elif command == "disable" and suffix == ".target":
-                    description = _("Cannot disable systemd target <code>{name!s}</code>.")
-                elif command == "mask":
-                    description = _("Cannot mask systemd unit <code>{name!s}</code>.")
-                else:
-                    description = _("Unknown systemd commands <code>{command!s}</code> and <code>{suffix!s}</code> for unit {name!s}.")
-
-                return (title,
-                        description.format(name=name, command=command, suffix=suffix) + " " + diagnostic
-                        )
+                title = _("Cannot modify unit")
+                diagnostic = _("<code>systemctl {_action!s}</code> call in chroot returned error code {_exit_code!s}.").format(_action=action, _exit_code=exit_code)
+                description = _("Cannot {_action!s} systemd unit <code>{_name!s}</code>.").format(_action=action, _name=name)
```

Comment:
I have some doubts that this will work well in translation, though. Translators will need to know that `{_action!s}` is not-to-be-translated, which is plausible. Also `enable`, `disable`, ... will end up as (latin1) text in the string, which might not make sense at all. In English it stands in for a verb, so it remains grammatical, but any other language is going to be quite pecuiliar.